plugins/Herald/plugin.py

```diff
@@ -87,31 +87,15 @@ def doQuit(self, irc, msg):
     def doJoin(self, irc, msg):
         if ircutils.strEqual(irc.nick, msg.nick):
             return # It's us.
-        if msg.nick in self.splitters:
-            self.log.debug('Not heralding %s, recent split.', msg.nick)
-            return # Recently split.
         channel = msg.args[0]
         irc = callbacks.SimpleProxy(irc, msg)
         if self.registryValue('heralding', channel, irc.network):
             try:
                 id = ircdb.users.getUserId(msg.prefix)
-                if id in self.splitters:
-                    self.log.debug('Not heralding id #%s, recent split.', id)
-                    return
-                herald = self.db[channel, id]
             except KeyError:
-                default = self.registryValue('default', channel, irc.network)
-                if default:
-                    default = ircutils.standardSubstitute(irc, msg, default)
-                    msgmaker = ircmsgs.privmsg
-                    if self.registryValue('default.notice',
-                                          channel, irc.network):
-                        msgmaker = ircmsgs.notice
-                    target = msg.nick
-                    if self.registryValue('default.public',
-                                          channel, irc.network):
-                        target = channel
-                    irc.queueMsg(msgmaker(target, default))
+                id = msg.nick
+            if id in self.splitters:
+                self.log.debug('Not heralding id #%s, recent split.', id)
                 return
```

Comment:
this change (using `id in self.splitters` instead of `msg.nick in self.splitters`) is not related to throttling, could you move it to a new PR?

Reply:
the reason throttle.afterpart doesn't work is because it only records parts for registered users, the fix only works because it now records msg.nick, so moving it to a new pull request is not really practical...

Reply:
Oh, I missed we already have `id in self.splitters`. nvm then